Forms/FormBuilder.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+namespace SilverStripe\Forms;
+
+use SilverStripe\Control\Controller;
+use SilverStripe\ORM\DataObject;
+
+/**
+ * A service which can generate a form for a given record and controller
+ */
+interface FormBuilder {
```

Comment:
I assume that we'd want to use this interface in CMSMain as well?

Calling it FormBuilder makes it sound like something tightly coupled to the React FormBuilder system. Perhaps `FormFactory`?